core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/common/accessor/ReflectionBeanPropertyMemberAccessor.java

```diff
@@ -15,15 +16,27 @@ public final class ReflectionBeanPropertyMemberAccessor extends AbstractMemberAc
     private final Class<?> propertyType;
     private final String propertyName;
     private final Method getterMethod;
+    private final MethodHandle getherMethodHandle;
     private final Method setterMethod;
+    private final MethodHandle setterMethodHandle;
 
     public ReflectionBeanPropertyMemberAccessor(Method getterMethod) {
         this(getterMethod, false);
     }
 
     public ReflectionBeanPropertyMemberAccessor(Method getterMethod, boolean getterOnly) {
         this.getterMethod = getterMethod;
-        getterMethod.setAccessible(true); // Performance hack by avoiding security checks
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
+        try {
+            getterMethod.setAccessible(true); // Performance hack by avoiding security checks
+            this.getherMethodHandle = lookup.unreflect(getterMethod)
```

Comment:
There is also `unreflectGetter` and `unreflectSetter`, but these operate on `Field` instances, while here we already have the method.

Reply:
Exactly, which is why I can't use those specialized methods here. There is a member accessor to deal specifically with fields.